resolver/dns/dns_resolver.go

```diff
@@ -111,6 +108,15 @@ func (b *dnsBuilder) Build(target resolver.Target, cc resolver.ClientConn, opts
 		disableServiceConfig: opts.DisableServiceConfig,
 	}
 
+	if target.Authority == "" {
+		d.resolver = defaultResolver
+	} else {
+		d.resolver, err = customAuthorityResolver(target.Authority)
+		if err != nil {
```

Comment:
Fallback to default resolver?

Reply:
I'd prefer the error. If I asked for a customer resolver there is likely a good reason I don't want to use the default one.